src/models.js

```diff
@@ -0,0 +1,188 @@
+class GraphFunc {
+    constructor (f='cos(2*x + 2*y)/(1 + x^2 + y^2)', rangeX=[-2, 2],
+                 rangeY=[-2, 2], meshX=10, meshY=10, res=30,
+                 lvlUp=true, viewGraph=true, rangeT=[0, 2], tangents=false,
+                 color='#000000', integrate=false, n=1, sample=0) {
+        this.f = f;
+        this.rangeX = rangeX;
+        this.rangeY = rangeY;
+        this.meshX = meshX;
+        this.meshY = meshY;
+        this.res = res;
+        this.lvlUp = lvlUp;
+        this.viewGraph = viewGraph;
+        this.rangeT = rangeT;
+        this.tangents = tangents;
+        this.color = color;
+        this.integrate = integrate;
+        this.n = n;
+        this.sample = sample;
+    }
+
+    type = 'GraphFunc';
```

Comment:
I think this `type` attribute may be redundant as we can get the class name of an instance with `obj.constructor.name`.